### ISSUE TITLE
[Submission] BackupTransport Enabler (Android P)

### ISSUE THREAD

**Issue by Mirr0ch1** [Magisk-Modules-Repo/submission#139]
https://github.com/Mirr0ch1/Magisk-BEP
I create this repo for the convenience of the Mainland version of OnePlus's H2OS firmware.
The README file is bilingually applied, came with both Chinese (simplified) and English.

**Comment by sjayminsgithub** [comment]
@Mirr0ch1 I will fully review your module and get back to you within a few hours. I hope you don't mind.

**Comment by Mirr0ch1** [comment]
yeah, sure, thanks.

**Comment by sjayminsgithub** [comment]
Well, some points I'd like to mention,
*) English should be first in README, or you could at least mention English is below Chinese description.
*) Your README shows incorrect information, if those files are already present, they will be overridden by module's files.
*) You must not include a reserve/ folder in system/ folder. This will Magic Mount them, which might render bootloops in many devices. You should locate the folder in root of your module.
*) Any module must perform compatibility checks on target device, in your case, API, device name and device ROM is what should be checked.

Please fix aforementioned issues in 48 hours, or I will have to reject the submission.

**Comment by Mirr0ch1** [comment]
Thanks, I will change the first and second issue right away.
One thing needs to mention is that this module is targeting OnePlus's H2OS (aka. China Mainland version of their Android firmware) as the top priority. And there's actually a `/system/reserve/` for storing all those bloatware shipped by OnePlus. I've been using this configuration on my OnePlus5 with H2OS installed for nearly a month, and there are no compatibility issues. Therefore I would like to ask you to remove the limitation stated under issue 3 and 4.

**Comment by Mirr0ch1** [comment]
Done, please review the source. Thanks for your work.
@sjayminsgithub

**Comment by sjayminsgithub** [comment]
Well, if a reserve folder actually exists in H2 OS, then fixing the 4th issue will automatically fix the third one. By compatibility checks, I meant that you should check whether the target device is a OnePlus 5 with H2 OS based on Android Pie. Such checks are required so that may prevent users from installing any module that's actually not of any use to them (and also from any damage).

**Comment by Mirr0ch1** [comment]
I'm sorry, but how to perform a compatibility test exactly? I didn't see any introduction under the document.

?? Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by Mirr0ch1** [comment]
Sorry for my dumb question above. I kinda get your point now. It won't be necessary because I will simply remove that directory.

**Comment by sjayminsgithub** [comment]
Sorry to disappoint you, but target device checks will still be necessary due to point 4 above. You can take a look at other modules, many of them have it (not all since they're universal).

**Comment by Mirr0ch1** [comment]
Thanks, but there's some point of views I would like to append:
The initiative of packing this module is to minimise the procedures it needs to take to bring Google Backup feature online after a clean flash of H2OS firmware. 
After that, I found out this approach works for H2OS 9.0 across devices -- which means OnePlus 5/5t/6/6t can all benefit from it. 
Furthermore, it is also a feasible approach for most of the Chinese version of Android vendor firmware to have Google Backup feature reinstalled quickly. Since most of which firmware actually get passed of the CTS test then get their GMS suite removed or hidden due to "regulation" in Mainland China. Which means MIUI from Xiaomi, ZUI from Lenovo, Flyme OS from Meizu and even Smartisan OS from Smartisan, are all compatible flashing this module.
What I want to say is that I don't think it's necessary to have a device model check function being built into the flash script. I've made a pretty clear statement about the modifications this module made under the README file. Such "power users" as the users are will certainly not do stupid things, e.g. flash this module on an Android O device. But I may perform an arm64 architecture check in the future.
